vic/vic_run/include/vic_log.h

```diff
@@ -45,51 +45,91 @@
 #include <errno.h>
 #include <string.h>
 
-FILE *LOG_DEST;
-
-#ifdef NDEBUG
-#define debug(M, ...)
-#else
-#define debug(M, ...) fprintf(LOG_DEST, "[DEBUG] %s:%d: " M "\n", __FILE__, \
-                              __LINE__, ## __VA_ARGS__)
+// Set the log level
+// To turn off debug statments, set LOG_LVL > 10
+// | Level     | Numeric value    |
+// |---------  |---------------   |
+// | ERROR     | 40               | > 40 will raise a compile time error
+// | WARNING   | 30               |
+// | INFO      | 20               |
+// | DEBUG     | 10               |
+// | NOTSET    | 0                |
+#ifndef LOG_LVL
+#define LOG_LVL 0
 #endif
 
+FILE *LOG_DEST;
+
 void finalize_logging(void);
 void get_current_datetime(char *cdt);
 void get_logname(const char *path, int id, char *filename);
 void initialize_log(void);
 void setup_logging(int id);
 
-// do not try to be smart and make this go away on NDEBUG, the _debug
-// here means that it just doesn't print a message, it still does the
-// check.  MKAY?
-#define check_debug(A, M, ...) if (!(A)) {debug(M, ## __VA_ARGS__); errno = 0; \
-                                          exit(1); }
-
+// Macros for logging
 #define clean_errno() (errno == 0 ? "None" : strerror(errno))
 #define clean_ncerrno(e) (nc_strerror(e))
 
+// Debug Level
+#if LOG_LVL > 10
+#define debug(M, ...)
+#else
+#define debug(M, ...) fprintf(LOG_DEST, "[DEBUG] %s:%d: " M "\n", __FILE__, \
+                              __LINE__, ## __VA_ARGS__)
+#endif
+
+// Info Level
+#if LOG_LVL > 20
+#define log_info(M, ...)
+#else
+#ifdef NO_LINENOS
+#define log_info(M, ...) fprintf(LOG_DEST, "[INFO] " M "\n", ## __VA_ARGS__)
+#else
+#define log_info(M, ...) fprintf(LOG_DEST, "[INFO] %s:%d: " M "\n", __FILE__, \
+                                 __LINE__, ## __VA_ARGS__)
+#endif
+#endif
+
+// Warn Level
+#if LOG_LVL > 30
+#define log_warn(M, ...)
+#else
+#ifdef NO_LINENOS
+#define log_warn(M, ...) fprintf(LOG_DEST, "[WARN] errno: %s: " M "\n", \
+                                 clean_errno(), ## __VA_ARGS__); errno = 0
+#else
+#define log_warn(M, ...) fprintf(LOG_DEST, "[WARN] %s:%d: errno: %s: " M "\n", \
+                                 __FILE__, __LINE__, \
+                                 clean_errno(), ## __VA_ARGS__); errno = 0
+#endif
+#endif
+
+// Error Level
+#if LOG_LVL > 40
+// Make sure we still raise an exit code
+#error "VIC Compile Time Error: LOG_LVL is set such that error messages will not be printed, however errors will still be raised"
```

Comment:
This error should just say that you are exiting because the error level is not defined (that is what you say in the commit comment)